packages/@aws-cdk/custom-resources/lib/provider-framework/runtime/cfn-response.ts

```diff
@@ -0,0 +1,47 @@
+// tslint:disable: max-line-length
+// tslint:disable: no-console
+import url = require('url');
+import { httpRequest } from './outbound';
+import { log } from './util';
+
+export interface CloudFormationResponseOptions {
+  readonly reason?: string;
+  readonly noEcho?: boolean;
+}
+
+export interface CloudFormationEventContext {
+  StackId: string;
+  RequestId: string;
+  PhysicalResourceId?: string;
+  LogicalResourceId: string;
+  ResponseURL: string;
+  Data?: any
+}
+
+export async function submitCloudFormationResponse(status: 'SUCCESS' | 'FAILED', event: CloudFormationEventContext, options: CloudFormationResponseOptions = { }) {
+  const json: AWSLambda.CloudFormationCustomResourceResponse = {
+    Status: status,
+    Reason: options.reason || status,
+    StackId: event.StackId,
+    RequestId: event.RequestId,
+    PhysicalResourceId: event.PhysicalResourceId || event.RequestId,
+    LogicalResourceId: event.LogicalResourceId,
+    NoEcho: options.noEcho,
+    Data: event.Data
+  };
+
+  log('submit response to cloudformation', json);
+
+  const responseBody = JSON.stringify(json);
+
+  const parsedUrl = url.parse(event.ResponseURL);
+  await httpRequest({
```

Comment:
Nit: no need to `await`, can just `return` this.

Reply:
I prefer to always use await for async functions to avoid the mistake of not calling await when it is required.